packages/esm-patient-appointments-app/src/appointments/appointments.resource.tsx

```diff
@@ -1,5 +1,62 @@
+import dayjs from 'dayjs';
+import useSWR from 'swr';
 import { openmrsFetch } from '@openmrs/esm-framework';
-import { Appointment, AppointmentPayload } from '../types';
+import { AppointmentPayload, AppointmentService, AppointmentsFetchResponse } from '../types';
+
+export const appointmentsSearchUrl = `/ws/rest/v1/appointments/search`;
+
+export function useAppointments(patientUuid: string, startDate: string, abortController: AbortController) {
+  /* 
+    SWR isn't meant to make POST requests for data fetching. This is a consequence of the API only exposing this resource via POST.
+    This works but likely isn't recommended.
+  */
+  const fetcher = () =>
+    openmrsFetch(appointmentsSearchUrl, {
+      method: 'POST',
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: {
+        patientUuid: patientUuid,
+        startDate: startDate,
+      },
+    });
+
+  const { data, error, isValidating } = useSWR<AppointmentsFetchResponse, Error>(appointmentsSearchUrl, fetcher);
+
+  const appointments = data?.data?.length
+    ? data.data.sort((a, b) => (b.startDateTime > a.startDateTime ? 1 : -1))
+    : null;
+
+  const pastAppointments = appointments?.filter((appointment) =>
+    dayjs((appointment.startDateTime / 1000) * 1000).isBefore(dayjs()),
```

Comment:
What's the math do?

Reply:
Hmmm, good question. Doesn't seem to do anything. I'll remove it in a future PR.